Code/Common/include/Ancillary/type_list2.h

```diff
@@ -0,0 +1,255 @@
+/*=========================================================================
+*
+*  Copyright NumFOCUS
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*         http://www.apache.org/licenses/LICENSE-2.0.txt
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+*=========================================================================*/
+#ifndef sitk_typelist_h_
+#define sitk_typelist_h_
+
+#include <tuple>
+#include <type_traits>
+#include <algorithm>
+
+namespace typelist2
+{
+
+/** \struct typelist
+ * \brief an non-instantiable holder of variadic template parameters to represent a list of types
+ *
+ * Can be used as a not and not an object. It enables complex compile-time manipulation of sets of types.
+ */
+template <typename... Ts>
+struct typelist;
```

Comment:
I find it a bit difficult to distinguish `typelist2` and `typelist`. Maybe follow the code convention of having namespace name in lower case, and template name in CamelCase? Like `typelist2::TypeList`?

Reply:
Yea, the naming is a bit horrible. `TypeList` was the name of the old class. It was easier to search and change to a different name. I think it will be each to change back to TypeList now.